packages/lib/onedrive-api.ts

```diff
@@ -339,6 +341,15 @@ export default class OneDriveApi {
 
 					await handleRequestRepeat(error);
 					continue;
+				} else if (error?.code === 'activityLimitReached' && response?.headers?._headers['retry-after'][0]) {
+					// Wait for OneDrive throttling
+					// Relavent Microsoft Docs: https://docs.microsoft.com/en-us/sharepoint/dev/general-development/how-to-avoid-getting-throttled-or-blocked-in-sharepoint-online#best-practices-to-handle-throttling
+					// Decrement retry count as multiple sync threads will cause repeated throttling errors - this will wait until throttling is resolved to continue, preventing a hard stop on the sync
+					i--;
```

Comment:
Hmm, if OneDrive keeps sending the activityLimitReached will end up in an infinite loop, although I'm not sure what else we can do about it. We can't set an upper limit I guess since on initial sync there could be many of these throttle errors.

Reply:
Indeed, I feel weird about this also. As I mentioned before, I assume given unlimited time/retries that the request will either succeed or result in a different error code which will then be handled normally. 

If you want to implement a failsafe we can try to think of an alternative. I tested with 20 threads and would get errors because they all would keep retrying but since there were multiple threads it kind of dragged the throttling longer for some threads which then exceeded the current 5 retry count and killed the sync completely.

Reply:
Not let's leave it as it is, and hopefully it won't be a problem. The fact that we're printing warnings to the log means we'll at least know what's happening.